doc/decisions/header_file_structure.md

```diff
@@ -12,31 +12,135 @@ The big problem is kdbprivate.h. It has two main problems:
 
 ## Constraints
 
+- Each library should have at least one separate header file:
+
+  Without this constraint, we would constantly include things that aren't needed.
+  This is inconvenient as it clutters the auto-suggestions of IDEs.
+  It also slows down the compile process, since the compiler has to parse everything that was included (see [also](https://lore.kernel.org/lkml/YdIfz+LMewetSaEB@gmail.com/T/)).
+
+- Non-public API and public API should be kept in separate files
+- Some non-public APIs need to be accessible for testing, such headers should not be packaged/installed
+
 ## Assumptions
 
+- There are different categories of "private":
+
+  1.  Some parts are truly private, i.e., shouldn't be used outside the library that defines them.
+      These things are only there because a library was split into multiple `.c` files.
+      This includes e.g., `splitNew` and `struct _Split`.
```

Comment:
Do they still exist?

Reply:
No, but `backendsMerge` and `struct _BackendData` do, I'll change that.